### PR TITLE
TSK-2838

### PR DESCRIPTION
**Pull request by JohannesDoberer**

### CODE REVIEW

File: src/main/java/com/hybris/builder/CLIRunner.java [hunk 1]
```diff
@@ -179,6 +190,12 @@ private void deleteRecursively(File path)
         path.delete();
     }
 
+    /**
+     * Creates a directory named of the argument and the ui-module files based on the simple module template.
```
Comment (hardl) [inline src/main/java/com/hybris/builder/CLIRunner.java]: not limited to builder modules

File: src/main/java/com/hybris/builder/CLIRunner.java [hunk 2]
```diff
@@ -260,11 +282,24 @@ protected String getString(InputStream inputStream) throws IOException
         return sb.toString();
     }
 
+    /**
+     * Show usage of the builder cli tool without an error Message.
```
Comment (hardl) [inline src/main/java/com/hybris/builder/CLIRunner.java]: remove "without an error message"

File: src/main/java/com/hybris/builder/CLIRunner.java [hunk 3]
```diff
@@ -260,11 +282,24 @@ protected String getString(InputStream inputStream) throws IOException
...
+     * @param command The command, which the builder cli should perform.
+     * @param error Boolean value, if an error occurred.
```
Comment (hardl) [inline src/main/java/com/hybris/builder/CLIRunner.java]: true, if an error notification should be printed out

File: src/main/java/com/hybris/builder/CLIRunner.java [hunk 4]
```diff
@@ -499,12 +564,19 @@ public String processCommand(String command, Map<String, String> depsMap)
         }
     }
 
+    /**
+     * Read the cmd and arg property of the command which will be executed. After all dependencies are resolved all commands saved in a List and call the real execute method.
```
Comment (hardl) [inline src/main/java/com/hybris/builder/CLIRunner.java]: ?

File: src/main/java/com/hybris/builder/CLIRunner.java [hunk 5]
```diff
@@ -521,6 +593,13 @@ public int execute(Properties cmdProps, String[] args, Map<String, String> depsM
         return 0;
     }
 
+    /**
+     * Execute the commandLine. See the constructor of {@ProccessBuilder}.
```
Comment (hardl) [inline src/main/java/com/hybris/builder/CLIRunner.java]: typo, ProcessBuilder

File: src/main/java/com/hybris/builder/CLIRunner.java [hunk 6]
```diff
@@ -671,11 +758,20 @@ protected void lookupMavenBinary() throws IOException
...
+     * Check if the operating system is a windows machine.
+     * @return true if the operating system is a windows machine.
+     */
     private boolean isWindows()
     {
         return System.getProperty("os.name").contains("indows");
     }
 
+    /**
+     * Check if maven binary is installed independent of the operating system.
```
Comment (hardl) [inline src/main/java/com/hybris/builder/CLIRunner.java]: not bound to maven binary

File: src/main/java/com/hybris/builder/CLIRunner.java [hunk 7]
```diff
@@ -671,11 +758,20 @@ protected void lookupMavenBinary() throws IOException
...
+     * @param filename
+     * @return the absolute path to the maven binary.
```
Comment (hardl) [inline src/main/java/com/hybris/builder/CLIRunner.java]: not bound to maven binary

Comment (Olga-Re) [inline src/main/java/com/hybris/builder/CLIRunner.java]: Machst du irgendwas mit denen? Sagen dir die Meldungen was?

**Pull request merged**